File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        newstr = ''
        for i in range(len(s)):
            if 'A'<=s[i]<='Z':
                newstr += s[i].lower()
            elif 'a'<=s[i]<='z':
                newstr += s[i]
            elif '0'<=s[i]<='9':
                newstr += s[i]
        newstr2 = ''
        for j in range(len(newstr)-1,-1,-1):
            newstr2 += newstr[j]
        if newstr == newstr2:
            return True
        else:
            return False
            

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String new_str = "";
        for (int i=0; i<s.length(); i++){
            if(Character.isLetterOrDigit(s.charAt(i))){
                new_str += Character.toLowerCase(s.charAt(i));
            }
        }
        int left = 0;
        int right = new_str.length() - 1;
        while (left<right){
            if (new_str.charAt(left) != new_str.charAt(right)){
                return false;
            }
            left ++;
            right --;
        }
        return true;
         
    }
}